### PR TITLE
Merge NCAR:dtc/develop into develop 2020/04/14

### PR DESCRIPTION
**Pull request by climbfuji**
- for macOS with LLVM clang, turn off OpenMP (from UFS public release)
- switch to Intel 18 on wcoss_cray (from @DusanJovic-NOAA)
- new regression tests with RRTMGP using current control as basis
- required cmake version now 3.0, set `project` to avoid annoying cmake warnings
- update GNU compiler on Cheyenne to 9.1.0 (required for RRTMGP, which uses Fortran 2008 standards that are not supported in GNU 8.3.0)
- build system improvement: get CCPP kind and type definitions from CCPP framework
- adjust formatting in `tests/rt.conf`
- make definition of RTPWD on cheyenne consistent with hera
- support for SECC's S4 (from @aerorahul), including update submodule pointer for NEMS -> NCEPLIBS-pyprodutil
- update of submodule pointer for WW3 to https://github.com/NOAA-EMC/WW3/commit/7ca6e5a623c90a0c26f416e921c484f8acfabc5d (required for using Intel 18 on wcoss_cray)

A new baseline was generated on hera, it can be found in
```
/scratch1/BMC/gmtb/ufs-weather-model/RT/NEMSfv3gfs/dtc-develop-20200413/INTEL
```
This baseline differs from the existing EMC baseline
```
/scratch1/NCEPDEV/nems/emc.nemspara/RT/NEMSfv3gfs/develop-20200323/INTEL
```
in the following:
- new input directory `FV3_input_data_RRTMGP`
- new regression test results `fv3_rrtmgp_ccpp`, `fv3_rrtmgp_debug_ccpp` and `fv3_rrtmgp_repro`
- updated regression test results for `fv3_gocart_clm_ccpp`, `fv3_gfsv16_csawmg_ccpp`, `fv3_gfsv16_csawmgt_ccpp` (all in PROD mode), because of the missing update of the `Diag%fluxr` quantities that was copied from IPD to CCPP when updating dtc/develop from develop and master; see comment here: https://github.com/NCAR/ufs-weather-model/pull/40#issuecomment-613116674

Associated PRs:

https://github.com/NOAA-EMC/GFDL_atmos_cubed_sphere/pull/17
https://github.com/NCAR/ccpp-framework/pull/283
https://github.com/NCAR/ccpp-physics/pull/436
https://github.com/NOAA-EMC/fv3atm/pull/98
https://github.com/ufs-community/ufs-weather-model/pull/102
https://github.com/NOAA-EMC/NEMS/pull/51

For regression testing information, see below.

### CODE REVIEW

File: tests/rt.conf [hunk 1]
```diff
@@ -379,19 +379,21 @@ RUN     | fv3_ccpp_thompson_no_aero
...
+RUN     | fv3_ccpp_rrtmgp                                                                                                                | standard    | hera.intel     | fv3         |
 
 #######################################################################################################################################################################################
 # CCPP DEBUG tests                                                                                                                                                                     #
 #######################################################################################################################################################################################
 
-COMPILE | CCPP=Y STATIC=Y SUITES=FV3_GFS_v15p2,FV3_GFS_v16beta DEBUG=Y                                                                   | standard    | hera.intel     | fv3         |
+COMPILE | CCPP=Y STATIC=Y SUITES=FV3_GFS_v15p2,FV3_GFS_v16beta,FV3_GFS_2017_RRTMGP DEBUG=Y                                               | standard    | hera.intel     | fv3         |
```
Comment (junwang-noaa) [inline tests/rt.conf:392]: Could we change all the rrtmgp test to be based on GFS_v16beta in later commits? We are going to update some of GFSv15 RT tests to run GFS_v16 configuration.

Comment (climbfuji) [reply tests/rt.conf:392]: Yes, certainly. Dustin did the original work based on fv3_ccpp_control.

**Comment by climbfuji** [comment]
Regression testing on cheyenne.gnu using `rt_gnu.conf` against existing baseline (generated previously from EMC develop and extended with RRTMG input data and regression test results): all tests pass. Since we generally don't update these log files in the PRs (not a tier-1 platform), I am attaching all the logs here.

[rt_cheyenne_gnu.log](https://github.com/ufs-community/ufs-weather-model/files/4477643/rt_cheyenne_gnu.log)
[Compile_cheyenne.gnu.log](https://github.com/ufs-community/ufs-weather-model/files/4477649/Compile_cheyenne.gnu.log)
[RegressionTests_cheyenne.gnu.log](https://github.com/ufs-community/ufs-weather-model/files/4477653/RegressionTests_cheyenne.gnu.log)

**Comment by climbfuji** [comment]
I did a diff of the old versus the new baseline (now in the usual EMC place with date tag 20200413) to make sure there are no other changes than those expected. Looks ok.
```
[Dom.Heinzeller@hfe08 INTEL]$ for dir in *; do diff -r $dir ../../develop-20200413/INTEL/$dir; done 2>&1 | tee $HOME/diff_EMC_baseline_20200323_20200413.log

Binary files fv3_gfsv16_csawmg_ccpp/phyf000.nemsio and ../../develop-20200413/INTEL/fv3_gfsv16_csawmg_ccpp/phyf000.nemsio differ
Binary files fv3_gfsv16_csawmg_ccpp/phyf024.nemsio and ../../develop-20200413/INTEL/fv3_gfsv16_csawmg_ccpp/phyf024.nemsio differ
Binary files fv3_gfsv16_csawmgt_ccpp/phyf000.nemsio and ../../develop-20200413/INTEL/fv3_gfsv16_csawmgt_ccpp/phyf000.nemsio differ
Binary files fv3_gfsv16_csawmgt_ccpp/phyf024.nemsio and ../../develop-20200413/INTEL/fv3_gfsv16_csawmgt_ccpp/phyf024.nemsio differ
Binary files fv3_gocart_clm_ccpp/phyf000.nemsio and ../../develop-20200413/INTEL/fv3_gocart_clm_ccpp/phyf000.nemsio differ
Binary files fv3_gocart_clm_ccpp/phyf024.nemsio and ../../develop-20200413/INTEL/fv3_gocart_clm_ccpp/phyf024.nemsio differ
```

I also did a diff "the other way round", i.e. the new versus the old baseline, and the results are also as expected:
```
[Dom.Heinzeller@hfe08 INTEL]$ for dir in *; do diff -r $dir ../../develop-20200323/INTEL/$dir; done 2>&1 | tee $HOME/diff_EMC_baseline_20200413_20200323.log
Binary files fv3_gfsv16_csawmg_ccpp/phyf000.nemsio and ../../develop-20200323/INTEL/fv3_gfsv16_csawmg_ccpp/phyf000.nemsio differ
Binary files fv3_gfsv16_csawmg_ccpp/phyf024.nemsio and ../../develop-20200323/INTEL/fv3_gfsv16_csawmg_ccpp/phyf024.nemsio differ
Binary files fv3_gfsv16_csawmgt_ccpp/phyf000.nemsio and ../../develop-20200323/INTEL/fv3_gfsv16_csawmgt_ccpp/phyf000.nemsio differ
Binary files fv3_gfsv16_csawmgt_ccpp/phyf024.nemsio and ../../develop-20200323/INTEL/fv3_gfsv16_csawmgt_ccpp/phyf024.nemsio differ
Binary files fv3_gocart_clm_ccpp/phyf000.nemsio and ../../develop-20200323/INTEL/fv3_gocart_clm_ccpp/phyf000.nemsio differ
Binary files fv3_gocart_clm_ccpp/phyf024.nemsio and ../../develop-20200323/INTEL/fv3_gocart_clm_ccpp/phyf024.nemsio differ
diff: ../../develop-20200323/INTEL/FV3_input_data_RRTMGP: No such file or directory
diff: ../../develop-20200323/INTEL/fv3_rrtmgp_ccpp: No such file or directory
diff: ../../develop-20200323/INTEL/fv3_rrtmgp_debug_ccpp: No such file or directory
diff: ../../develop-20200323/INTEL/fv3_rrtmgp_repro: No such file or directory
```

**Comment by climbfuji** [comment]
Regression testing on cheyenne.intel using `rt_intel.conf` (copied from dtc/develop branch, doesn't exist in develop / master) against existing baseline (generated previously from dtc/develop): all tests pass. Since we generally don't update these log files in the PRs (not a tier-1 platform), I am attaching all the logs here.

[Compile_cheyenne.intel.log.gz](https://github.com/ufs-community/ufs-weather-model/files/4481458/Compile_cheyenne.intel.log.gz)
[RegressionTests_cheyenne.intel.log](https://github.com/ufs-community/ufs-weather-model/files/4481460/RegressionTests_cheyenne.intel.log)
[rt_cheyenne_intel.log](https://github.com/ufs-community/ufs-weather-model/files/4481461/rt_cheyenne_intel.log)

**Comment by climbfuji** [comment]
Regression testing on hera.intel using NEMSCompsetRun: all tests pass, logs updated in the PR.

[NEMSCompsetRun_hera_intel.log](https://github.com/ufs-community/ufs-weather-model/files/4483403/NEMSCompsetRun_hera_intel.log)

**Comment by climbfuji** [comment]
Update on the regression test baseline: due to the WW3 submodule pointer update, the WW3_input_data directory as well as the results of the regression tests with WW3=Y were updated:
```
WW3_input_data
fv3_gfdlmprad
fv3_gfdlmprad_atmwav
fv3_wrtGauss_nemsio_c768
fv3_gfdlmprad_repro
fv3_gfdlmprad_atmwav_repro
fv3_wrtGauss_nemsio_c768_repro
fv3_gfdlmprad_ccpp
fv3_gfdlmprad_atmwav_ccpp
fv3_wrtGauss_nemsio_c768_ccpp
```

**Comment by climbfuji** [comment]
Regression testing on hera.intel using rt.sh: all tests pass, logs updated in the PR.

[rt_hera_intel.log](https://github.com/ufs-community/ufs-weather-model/files/4483694/rt_hera_intel.log)

**Comment by climbfuji** [comment]
Regression testing on wcoss_dell_p3 using NEMSCompsetRun and rt.sh: all tests pass, logs updated in the PR.

**Comment by climbfuji** [comment]
Regression testing on wcoss_cray using NEMSCompsetRun and rt.sh: all tests pass, logs updated in the PR.

**Caveat:** As described in issue https://github.com/ufs-community/ufs-weather-model/issues/103, the gmake build uses the wrong flags for the REPRO mode. This build is used when invoking `./rt.sh ...` without ecflow or rocoto, while the ecflow/rocoto versions `./rt.sh -e ...` and `./rt.sh -c ...` use the cmake build. The baseline on wcoss cray was created/updated with the results of the cmake build (from `./rt.sh -e ...`) and thus is correct. Once the changes described in https://github.com/ufs-community/ufs-weather-model/issues/103 are made, the gmake build should give the same results.

**Comment by climbfuji** [comment]
These PRs are ready for merging from my side.

**Review by junwang-noaa — APPROVED** [review]

**Review by DusanJovic-NOAA — APPROVED** [review]

**Pull request merged**